### PR TITLE
mathlib Limits don't bother using references

### LINKED ISSUE #11408: Issue with Master_vehicle crash
Today we tested Master v1.9.0 (beta) (7f3b1700) with the Pixracer, Pixhawk 4 mini, Pixhawk 4. Created mission. The vehicle armed good, took off. At 30 meters the vehicle started spinning and came down to the ground. Resulted in a crash. Same behavior for all 3 platforms. 

1. Uploaded mission
1. Drone switched on
1. Vehicle armed
1. Took off
1. Elevated to 30 meters
1. vehicle started spinning out of control
1. The vehicle came down and crashed

Pixracer:
https://review.px4.io/plot_app?log=4f0c2692-5f27-4d83-bfbd-765bf48a989e

Pixhawk 4 Mini:
https://review.px4.io/plot_app?log=53fa724b-906d-4e93-87fe-5a4f4c2a50b8

Pixhawk 4:
https://review.px4.io/plot_app?log=69f42b00-6259-4a97-96c1-bd8f06b4d29d

@dagar

### PR DESCRIPTION
**Pull request by dagar**
- cleanup Limits.hpp
 - add -Waddress-of-packed-member where available
 - fixes #11408

### CODE REVIEW

File: platforms/nuttx/cmake/px4_impl_os.cmake [hunk 1]
```diff
@@ -72,6 +72,8 @@ function(px4_os_add_flags)
 	# prevent using the toolchain's std c++ library
 	add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-nostdinc++>)
 
+	add_compile_options(-mno-unaligned-access)
```
Comment (bkueng) [inline platforms/nuttx/cmake/px4_impl_os.cmake]: Why did you add this?

Comment (dagar) [reply platforms/nuttx/cmake/px4_impl_os.cmake]: Considering other possible alignment issues. Removing from this PR.

**Comment by dagar** [comment]
image
